apps/ordermoments/order_moments.star

```diff
@@ -0,0 +1,749 @@
+            print("store_latest_celebration Error ❌: Status code %d, URL %s, Body %s" % (response.status_code, response.url, response.body()))
+            return False
+
+        return True
+
+# GET ORDER COUNT
+# Gets a number of orders for a provided store name using a provided API token
+# -----------------------------------------------------------------------------------------
+# store_name: A store name
+# api_token: An API token
+# Returns: A number representing the order count for a store, or -1 if the count couldn't be fetched
+def get_order_count(store_name, api_token):
+    # Check our cache
+    cache_key = CACHE_ID_ORDERS.format(hash.sha1(store_name))
+    cached_orders = cache.get(cache_key)
+
+    if not cached_orders:
+        # Nothing was in the cache, so fetch our orders now
+        url = REST_ENDPOINT.format(store_name, COUNT_ENDPOINT)
+        headers = {"Content-Type": "application/json", "X-Shopify-Access-Token": api_token}
+        response = http.get(url = url, params = {"status": "any"}, headers = headers)
+
+        # If there was any error, return -1
+        if response.status_code != 200:
+            print("get_order_count Error ❌: Status code %d, URL %s, Body %s" % (response.status_code, response.url, response.body()))
+            return -1
+
+        # Store the new value in cache
+        print("Cache 💾: Storing value for key %s" % cache_key)
+        cache.set(cache_key, response.body(), ttl_seconds = CACHE_TTL)
+        order_count = response.json()
+
+    else:
+        # Use our cached value
+        print("Cache 💾: Found cached value for key %s" % cache_key)
+        order_count = json.decode(cached_orders)
+
+    # Return our count value
+    return order_count["count"]
+
+# Error View
+# Renders an error message
+# -----------------------------------------------------------------------------------------
+# message: A message to display as a rendered error
+# Returns: A Pixlet root element
+def error_view(message):
+    return render.Root(
+        render.Stack(
+            children = [
+                render.Image(STARFIELD),
+                render.Column(
+                    expanded = True,
+                    main_align = "space_evenly",
+                    cross_align = "center",
+                    children = [
+                        render.Image(ALIEN_ERROR),
+                        render.Marquee(
+                            width = 64,
+                            offset_start = 64,
+                            child = render.Text(content = message, color = "#FF0"),
+                        ),
+                    ],
+                ),
+            ],
+        ),
+    )
+
+# Get Schema
+# Return a Pixlet Schema for this Celebrate Applet
+# -----------------------------------------------------------------------------------------
+# Returns: A Pixlet schema
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
```

Comment:
Any chance this could be OAuth2? If you're installing this app through the API, you can still override the config value. The benefit is that existing users of Tidbyt who have a Shopify store can discover this app and install it without having to jump through hoops setting up their API key.